content/docs/user-guide/gems/reference/multiplayer/multiplayer/tutorial_multiplayer.md

```diff
@@ -0,0 +1,887 @@
+{{</important>}}
+
+![Attaching Network Component to an Autonomous Entity](/images/user-guide/multiplayer/add_myfirstnetworkcomponent_to_player_prefab.png)
+
+Here is our full source code for our component with RPC. For the header, `D:\git\MultiplayerSample\Gem\Code\Source\Components\MyFirstNetworkComponent.h`:
+
+```c++
+#pragma once
+
+#include <Source/AutoGen/MyFirstNetworkComponent.AutoComponent.h>
+#include <AzCore/Component/TickBus.h>
+
+namespace MultiplayerSample
+{
+    class MyFirstNetworkComponent
+        : public MyFirstNetworkComponentBase
+    {
+    public:
+        AZ_MULTIPLAYER_COMPONENT( MultiplayerSample::MyFirstNetworkComponent, s_myFirstNetworkComponentConcreteUuid, MultiplayerSample::MyFirstNetworkComponentBase );
+
+        static void Reflect( AZ::ReflectContext* context );
+
+        MyFirstNetworkComponent();
+
+        void OnInit() override {}
+        void OnActivate( Multiplayer::EntityIsMigrating entityIsMigrating ) override;
+        void OnDeactivate( Multiplayer::EntityIsMigrating entityIsMigrating ) override;
+
+    private:
+        AZ::Event<double>::Handler m_uptimeChanged;
+        void OnUpTimeChanged( double uptime );
+    };
+
+    class MyFirstNetworkComponentController
+        : public MyFirstNetworkComponentControllerBase
+        , public AZ::TickBus::Handler
+    {
+    public:
+        MyFirstNetworkComponentController(MyFirstNetworkComponent& parent);
+
+        void OnActivate(Multiplayer::EntityIsMigrating entityIsMigrating) override;
+        void OnDeactivate(Multiplayer::EntityIsMigrating entityIsMigrating) override;
+
+        void OnTick( float deltaTime, AZ::ScriptTimePoint time ) override;
+
+        void HandleSendConfirmUptime(AzNetworking::IConnection* invokingConnection, const double& UpTime) override;
+    };
+}
+```
+
+The source `D:\git\MultiplayerSample\Gem\Code\Source\Components\MyFirstNetworkComponent.cpp`:
+```c++
+#include <Source/Components/MyFirstNetworkComponent.h>
+
+#include <AzCore/Serialization/SerializeContext.h>
+#include <Multiplayer/Components/NetBindComponent.h>
+
+namespace MultiplayerSample
+{
+    void MyFirstNetworkComponent::Reflect(AZ::ReflectContext* context)
+    {
+        AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serializeContext)
+        {
+            serializeContext->Class<MyFirstNetworkComponent, MyFirstNetworkComponentBase>()
+                ->Version(1);
+        }
+        MyFirstNetworkComponentBase::Reflect(context);
+    }
+
+    MyFirstNetworkComponent::MyFirstNetworkComponent()
+        : m_uptimeChanged([this](double uptime) { OnUpTimeChanged(uptime); })
+    {
+        if (IsNetEntityRoleAuthority())
+        {
+            GetNetBindComponent()->SetAllowAutonomy(true);
```

Comment:
this isn't required if this component is attached to the player prefab right? you already mention above to be sure to attach this component to the player.prefab, so i think we need to remove this line?

Reply:
Good point, removing.